src/NUnitEngine/nunit.engine/Drivers/NUnit3FrameworkDriver.cs

```diff
@@ -120,7 +121,24 @@ public string Run(ITestEventListener listener, string filter)
             CallbackHandler handler = new RunTestsCallbackHandler(listener);
 
             log.Info("Running {0} - see separate log file", Path.GetFileName(_testAssemblyPath));
-            CreateObject(RUN_ACTION, _frameworkController, filter, handler);
+
+            try
+            {
+                CreateObject(RUN_ACTION, _frameworkController, filter, handler);
+            }
+            catch (System.Reflection.TargetInvocationException ex)
+            {
+                string innerMessage = ex.InnerException.Message;
+                if (ex.InnerException is ArgumentException && innerMessage.StartsWith("Invalid filter element"))
```

Comment:
It looks like the `InnerException` contains enough information to help the user diagnose the problem, it is just the fact that a `TargetInvocationException` is obscuring the inner message. Instead of matching on the message, could we not instead just throw an `NUnitEngineException` with the message from any `InnerException`?

```C#
catch (TargetInvocationException ex)
{
    if (ex.InnerException != null)
    {
        throw new NUnitEngineException(ex.InnerException.Message);
    }
   throw;
}
```